lib/MiLightState/GroupState.cpp

```diff
@@ -462,6 +488,43 @@ void GroupState::applyField(JsonObject& partialState, GroupStateField field) {
   }
 }
 
+// helper function to debug the current state (in JSON) to the serial port
+void GroupState::debugState(char const *debugMessage) {
+#ifdef DEBUG_PRINTF
+  // using static to keep large buffers off the call stack
```

Comment:
I think we might want to not use `static` here.  I think that this will allocate 2KBs of RAM globally, instead of just when this method is called.  We really only have 10-20 KBs of RAM to play with before things start getting weird.